packages/core/middleware/src/helpers.ts

```diff
@@ -0,0 +1,18 @@
+export const getAgnosticStatusCode = (obj: unknown, ...keys: string[]) => {
+  if (Array.isArray(obj)) {
+    for (const o of obj) {
+      if (getAgnosticStatusCode(o, ...keys)) {
+        return getAgnosticStatusCode(o, ...keys);
+      }
+    }
+  } else if (obj && typeof obj === 'object') {
+    for (const key of Object.keys(obj)) {
+      if (keys.includes(key)) {
+        return obj[key];
+      } else if (getAgnosticStatusCode(obj[key], ...keys)) {
+        return getAgnosticStatusCode(obj[key], ...keys);
```

Comment:
I don't like it is called twice. Could you assign it to some variable and call only once?